### DISCUSSION TITLE
Build and development on Windows OS

### DISCUSSION THREAD

**Discussion by ohartvic** [OpenOrienteering/mapper#2254] [Questions & Answers] [answered]
I would like to start contributing to mapper. I am failing to build the project on Windows. I was following instructions in the INSTALL.md, unsuccesfully. 

Can someone please help?

**Comment by dg0yt** [comment]
Welcome, and thanks for asking. Can you tell us more about the specific problem? I do not use a Windows environment very often (for Mapper development).

**Comment by ohartvic** [comment]
Hi Kai, thanks for your reply. I can successfully run cmake. When running
build step this is where I end:

[image: image.png]

Best Regards,

Otto

**Reply by ohartvic** [reply]
... sorry, somehow lost the screenshot, see below:
![image](https://github.com/OpenOrienteering/mapper/assets/19303491/476022e0-a138-4c0d-8746-c99b87d08361)

The issue might have something to do with Windows OS, screenshot above is from Windows11 environment. I tried on two different computers, installing msys2mingw64 from scratch and no luck.

On Windows 10 it works - also tried on two different environments. So I am going to stick to W10 for the time being.

**Reply by dg0yt** [reply, marked as answer]
The failing `code-check-wrapper.sh` isn't really tuned for Windows. Either you completely disable the use of include-what-you-use and clang-tidy with these CMake options:
~~~
-DCMAKE_DISABLE_FIND_PACKAGE_ClangTidy=ON
-DCMAKE_DISABLE_FIND_PACKAGE_IWYU=ON
~~~
or you use the tools without that wrapper, e.g. for above error:
~~~
-DCMAKE_CXX_INCLUDE_WHAT_YOU_USE=C:/msys64/mingw64/bin/include-what-you-use.exe
~~~

**Reply by ohartvic** [reply]
Thank you, Kai.
